### PR TITLE
[RHCLOUD-26223] Fix single email for multiple recipients

### PR DESCRIPTION
**Pull request by g-duval**

### CODE REVIEW

File: engine/src/main/java/com/redhat/cloud/notifications/processors/email/EmailSubscriptionTypeProcessor.java [hunk 1]
```diff
@@ -164,8 +167,10 @@ private void sendEmail(Event event, Set<Endpoint> endpoints) {
                 .getEmailSubscribersUserId(event.getOrgId(), bundleName, applicationName, eventTypeName, emailSubscriptionType));
 
         Set<User> userList = recipientResolver.recipientUsers(event.getOrgId(), requests, subscribers);
-        if (featureFlipper.isSendSingleEmailForMultipleRecipientsEnabled()) {
-            emailSender.sendEmail(userList, event, subject, body, true);
+        if (isSendSingleEmailForMultipleRecipientsEnabled(userList)) {
+            if (!userList.isEmpty()) {
```
Comment (josejulio) [inline engine/src/main/java/com/redhat/cloud/notifications/processors/email/EmailSubscriptionTypeProcessor.java:171]: good catch!

Comment (josejulio) [reply engine/src/main/java/com/redhat/cloud/notifications/processors/email/EmailSubscriptionTypeProcessor.java:171]: I'm wondering if this check should be inside the `sendEmail` method

File: engine/src/main/java/com/redhat/cloud/notifications/processors/email/EmailSubscriptionTypeProcessor.java [hunk 2]
```diff
@@ -279,4 +284,12 @@ private void processAggregateEmailsByAggregationKey(EmailAggregationKey aggregat
             emailAggregationRepository.purgeOldAggregation(aggregationKey, endTime);
         }
     }
+
+    private boolean isSendSingleEmailForMultipleRecipientsEnabled(Set<User> users) {
+        if (ProfileManager.getLaunchMode() == NORMAL && featureFlipper.isSendSingleEmailForMultipleRecipientsEnabled()) {
+            Set<String> strUsers = users.stream().map(User::getUsername).collect(Collectors.toSet());
+            return (strUsers.contains("gduval-prod") || strUsers.contains("gduval-stage"));
```
Comment (josejulio) [inline engine/src/main/java/com/redhat/cloud/notifications/processors/email/EmailSubscriptionTypeProcessor.java:291]: I wonder if we should extend our feature flipper to be able to conditionally enable features on specific accounts. Just wanting to start a discussion and I don't think it should affect this PR

File: engine/src/main/java/com/redhat/cloud/notifications/processors/email/EmailSender.java [hunk 3]
```diff
@@ -55,6 +55,9 @@ public class EmailSender {
     @ConfigProperty(name = "processor.email.bop_env")
     String bopEnv;
 
+    @ConfigProperty(name = "processor.email.no_reply")
```
Comment (josejulio) [inline engine/src/main/java/com/redhat/cloud/notifications/processors/email/EmailSender.java:58]: This is already part of the clowder file: https://github.com/RedHatInsights/notifications-backend/blob/master/.rhcicd/clowdapp-engine.yaml#L369C9-L371

**Review by josejulio — APPROVED** [review]
lgtm

File: engine/src/main/java/com/redhat/cloud/notifications/processors/email/EmailSubscriptionTypeProcessor.java (hunk 2 above)
Comment (gwenneg) [reply engine/src/main/java/com/redhat/cloud/notifications/processors/email/EmailSubscriptionTypeProcessor.java:289]: That's one of the many ideas I wish I had time to implement in the quarkus-feature-flag extension (which doesn't exist yet) 😄

**Pull request merged**